x-pack/plugins/apm/common/projections/services.ts

```diff
@@ -7,37 +7,46 @@
 import {
   Setup,
   SetupUIFilters,
-  SetupTimeRange
+  SetupTimeRange,
+  SetupHasTransactionDurationMetrics
   // eslint-disable-next-line @kbn/eslint/no-restricted-paths
 } from '../../server/lib/helpers/setup_request';
-import { SERVICE_NAME, PROCESSOR_EVENT } from '../elasticsearch_fieldnames';
+import { SERVICE_NAME } from '../elasticsearch_fieldnames';
 // eslint-disable-next-line @kbn/eslint/no-restricted-paths
 import { rangeFilter } from '../../server/lib/helpers/range_filter';
 
 export function getServicesProjection({
   setup
 }: {
-  setup: Setup & SetupTimeRange & SetupUIFilters;
+  setup: Setup &
+    SetupTimeRange &
+    SetupUIFilters &
+    SetupHasTransactionDurationMetrics;
 }) {
-  const { start, end, uiFiltersES, indices } = setup;
+  const {
+    start,
+    end,
+    uiFiltersES,
+    indices,
+    hasTransactionDurationMetrics
+  } = setup;
+
+  const index = [
+    indices['apm_oss.metricsIndices'],
+    indices['apm_oss.errorIndices']
+  ];
+
+  if (!hasTransactionDurationMetrics) {
+    index.push(indices['apm_oss.transactionIndices']);
+  }
 
   return {
-    index: [
-      indices['apm_oss.metricsIndices'],
-      indices['apm_oss.errorIndices'],
-      indices['apm_oss.transactionIndices']
-    ],
+    index,
     body: {
       size: 0,
       query: {
         bool: {
-          filter: [
-            {
-              terms: { [PROCESSOR_EVENT]: ['transaction', 'error', 'metric'] }
-            },
```

Comment:
Removing the processor.event filter could cause problems. By default APM app queries `apm-*` which is all indicies. So this query will return transactions, metrics, errors and spans by default.
I think you'll need the same logic as for the index.

I think the queried indicies and processor.event should always be aligned. I was hoping we could add the `processor.event` filter in the setup (middleware) but we might be better with an explicit helper. Something like:

```js
const { index, processorEvent } = hasTransactionDurationMetrics
  ? // transactions are stored as pre-aggregated (histogram) docs in metric index
    getIndexAndProcessorEvent('metric', 'error')

  : // transactions are not pre-aggregated and stored in transaction index
        getIndexAndProcessorEvent('metric', 'error', transaction);
```